lib/spack/spack/cmd/repo.py

```diff
@@ -83,7 +88,22 @@ def repo_create(args):
 def repo_add(args):
     """Add a package source to Spack's configuration."""
     path = args.path
-
+    fetcher = fs.from_url(path)
+    spack_repo_config = spack.config.get_config('repos')
+    root_repo = Repo(spack_repo_config[-1])
+    if isinstance(fetcher, fs.VCSFetchStrategy):
+        with Stage("testpath") as stage:
+            fetcher.set_stage(stage)
+            fetcher.fetch()
+            packageName = os.listdir(stage.path)[0]
+            try:
+                shutil.rmtree(root_repo.packages_path + "/" + packageName)
+            except OSError:
+                pass
+            shutil.move(
+                stage.path + "/" + packageName,
+                root_repo.packages_path)
+            path = root_repo.packages_path + "/" + packageName
```

Comment:
All of these are written in Python-centric python, and should probably be in more of a spack idiom. Curious to know the correct way to say this.